tests/acceptance/helpers.py

```diff
@@ -163,22 +136,39 @@ def create_and_validate_request_and_response(endpoint, method, session, bypass_v
     :return:
         - response: API response object
     """
-    request, request_result = create_and_validate_request(
-        endpoint, method, payload, params, dict(session.headers)
+    base_url = os.getenv('host')
+    headers = dict(session.headers)
+    headers["Content-Type"] = "application/json"
+
+    request = Request(
+        url=base_url + endpoint,
+        method=method,
+        params=params,
+        data=payload,
+        headers=headers,
     )
 
-    if not bypass_validation_request:
-        request_result.raise_for_errors()
 
-    base_url = os.getenv('host')
+  
+    try:
+        create_and_validate_request(request)
+    except Exception as e:
+        if not bypass_validation_request:
+            raise e
 
     if method == 'post':
         response = session.post(base_url + endpoint, params=params, data=payload)
     elif method == 'get':
         response = session.get(base_url + endpoint, params=params, data=payload)
-    response_result = create_and_validate_response(request, response)
-
-    if not bypass_validation_response:
-        response_result.raise_for_errors()
 
+    try: 
+        create_and_validate_response(request, response)
+    except MissingData:
+        return response
+    except Exception as e:
```

Comment:
Same here. Would be easier to debug as error messages would be more specific. An example here: https://github.com/optimizely/python-sdk/blob/master/optimizely/odp/odp_segment_api_manager.py

But if that's too complicated we could just keep a generic Exception, this code is not super critical or super customer facing.

Reply:
same here as well, updated with specific exceptions